sdks/java/extensions/sql/zetasql/src/main/java/org/apache/beam/sdk/extensions/sql/zetasql/translation/ExpressionConverter.java

```diff
@@ -1000,9 +1002,19 @@ private static boolean isZetaSQLCast(TypeKind fromType, TypeKind toType) {
   }
 
   private RexNode convertResolvedParameter(ResolvedParameter parameter) {
-    assert parameter.getType().equals(queryParams.get(parameter.getName()).getType());
-    return convertValueToRexNode(
-        queryParams.get(parameter.getName()).getType(), queryParams.get(parameter.getName()));
+    if (queryParams.getKind() == Kind.NAMED) {
+      Map<String, Value> queryParameterMap = (Map<String, Value>) queryParams.named();
+      Value value = queryParameterMap.get(parameter.getName());
+      assert parameter.getType().equals(value.getType());
```

Comment:
If I understand correctly, this might be best as a `checkState`. That way you can attach a good error message and indicate that the internal state has gotten messed up.

Reply:
Done.